### PR TITLE
Disjoint Sets: Quick Union implementation

### PR DESCRIPTION
**Pull request by antonioaversa**

### CODE REVIEW

File: MoreStructures/DisjointSets/QuickUnionDisjointSet.cs [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    ///       is O(n).
+    ///       <br/>
+    ///     - Therefore, Time Complexity is O(n). Space Complexity is O(1).
+    ///     </para>
+    /// </remarks>
+    public bool AreConnected(int first, int second)
+    {
+        if (ValuesCount == 0)
+            throw new InvalidOperationException(
+                $"{nameof(AreConnected)} cannot be invoked on an empty queue.");
```
Comment (andreaguarino) [inline MoreStructures/DisjointSets/QuickUnionDisjointSet.cs:144]: did you mean `set`?

File: MoreStructures/DisjointSets/QuickUnionDisjointSet.cs [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    ///       is O(n) in the worst case.
+    ///       <br/>
+    ///     - Therefore, Time Complexity is O(n). Space Complexity is O(1).
+    ///     </para>
+    /// </remarks>
+    public int Find(int value)
+    {
+        if (ValuesCount == 0)
+            throw new InvalidOperationException(
+                $"{nameof(AreConnected)} cannot be invoked on an empty queue.");
```
Comment (andreaguarino) [inline MoreStructures/DisjointSets/QuickUnionDisjointSet.cs:184]: queue -> set

File: MoreStructures/DisjointSets/QuickUnionDisjointSet.cs [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    ///       parent in the list of parents, which is O(1) work.
+    ///       <br/>
+    ///     - Therefore, Time Complexity is O(n). Space Complexity is O(1).
+    ///     </para>
+    /// </remarks>
+    public void Union(int first, int second)
+    {
+        if (ValuesCount == 0)
+            throw new InvalidOperationException(
+                $"{nameof(AreConnected)} cannot be invoked on an empty queue.");
```
Comment (andreaguarino) [inline MoreStructures/DisjointSets/QuickUnionDisjointSet.cs:225]: queue -> set

**Review by andreaguarino — DISMISSED** [review]
LGTM, algorithm to count sets could be simplified

File: MoreStructures/DisjointSets/QuickUnionDisjointSet.cs (hunk 3 above)
Comment (antonioaversa) [reply MoreStructures/DisjointSets/QuickUnionDisjointSet.cs]: Fixed

File: MoreStructures/DisjointSets/QuickUnionDisjointSet.cs (hunk 2 above)
Comment (antonioaversa) [reply MoreStructures/DisjointSets/QuickUnionDisjointSet.cs]: Changed to set.

File: MoreStructures/DisjointSets/QuickUnionDisjointSet.cs (hunk 1 above)
Comment (antonioaversa) [reply MoreStructures/DisjointSets/QuickUnionDisjointSet.cs]: Yes, fixed.

File: MoreStructures/DisjointSets/QuickUnionDisjointSet.cs [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+                    processed.Add(root);
+                    root = Parents[root];
+                }
+
+                if (newRootFound)
+                    roots.Add(root);
+
+                processed.Add(i);
+            }
+            return roots.Count;
```
[the remark this answers is not in the document]
Comment (antonioaversa) [reply MoreStructures/DisjointSets/QuickUnionDisjointSet.cs]: Indeed! Improved in https://github.com/antonioaversa/MoreStructures/pull/173/commits/71ec8f93b826eb282ffc0d3ab3b0af456293225a#diff-577276d235a180a7c3dcb0425a2fa0786635634f79c9e8143059be61b99e3e26R68

**Comment by antonioaversa** [comment]
> LGTM, algorithm to count sets could be simplified

All comments addressed, algorithm to retrieve the number of sets simplified following the suggestion.

**Pull request merged**